pkg/cmd/cli/cmd/set/volume.go

```diff
@@ -532,6 +549,8 @@ func (v *VolumeOptions) getVolumeUpdatePatches(infos []*resource.Info, singular
 }
 
 func setVolumeSourceByType(kv *kapi.Volume, opts *AddVolumeOptions) error {
+	defaultMode, _ := strconv.ParseInt(opts.DefaultMode, 8, 32)
+	defaultMode32 := int32(defaultMode)
```

Comment:
> We then run into the issue of the kapi types, which are defined with *int32. Is there a way I can force Go to use os.Filetype?

`os.FileMode` is a `uint32` https://golang.org/pkg/os/#FileMode
Unless I misunderstood your comment, you should just be able to do something like:
```go
defaultMode32 := os.FileMode(defaultMode)
```